### PR TITLE
Feature/music service

### PR DESCRIPTION
**Pull request by AntonZhidovich**
Music service on clean architecture for working with playlists, track playback and viewing its information

MS SQL Server for storing information about authors, releases and songs.
MongoDB for storing user playlists and tariff plan constraints.
MinIO for storing soundtracks.

### CODE REVIEW

File: MusicService/MusicService.API/Controllers/GenreController.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        private readonly IGenreService _songService;
+
+        public GenreController(IGenreService songService)
+        {
+            _songService = songService;
+        }
+
+        [HttpGet("")]
+        public async Task<IActionResult> GetAllGenresAsync([FromQuery] GetPageRequest request)
```
Comment (katyalys) [inline MusicService/MusicService.API/Controllers/GenreController.cs:23]: why do you use "" in HttpGet?

File: MusicService/MusicService.Application/Mapping/AuthorMappingProfile.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+namespace MusicService.Application.Mapping
+{
+    public class AuthorMappingProfile : Profile
+    {
+        public AuthorMappingProfile()
+        {
+            CreateMap<Author, AuthorDto>()
+               .ForMember(dest => dest.UserNames, options => options.MapFrom(src => src.Users.Select(user => user.UserName)));
+
+            CreateMap<CreateAuthorRequest, Author>()
```
Comment (katyalys) [inline MusicService/MusicService.Application/Mapping/AuthorMappingProfile.cs:20]: same as in previous pr

File: MusicService/MusicService.Application/Services/AuthorService.cs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+            }
+
+            if (user.Author != null)
+            {
+                throw new BadRequestException(ExceptionMessages.UserAlreadyInAuthor);
+            }
+
+            var author = await GetDomainAuthorAsync(request.AuthorName, cancellationToken);
+
+            if (!currentUser.IsInRole(UserRoles.admin)) { CheckIfUserIsMember(author, currentUser); }
```
Comment (katyalys) [inline MusicService/MusicService.Application/Services/AuthorService.cs]: move to new line CheckIfUserIsMember(author, currentUser);

File: MusicService/MusicService.Application/Services/AuthorService.cs [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+            await _unitOfWork.CommitAsync(cancellationToken);
+
+            return _mapper.Map<AuthorDto>(author);
+        }
+
+        public async Task DeleteAsync(string name, ClaimsPrincipal currentUser, CancellationToken cancellationToken = default)
+        {
+            var author = await GetDomainAuthorAsync(name, cancellationToken);
+
+            if (!currentUser.IsInRole(UserRoles.admin)) { CheckIfUserIsMember(author, currentUser); }
```
Comment (katyalys) [inline MusicService/MusicService.Application/Services/AuthorService.cs]: new line for CheckIfUserIsMember(author, currentUser);

File: MusicService/MusicService.Application/Services/ReleaseService.cs [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+                    break;
+            }
+        }
+
+        private void CheckIfUserIsMember(IEnumerable<Author> authors, ClaimsPrincipal user)
+        {
+            var currentUserName = user.Identity!.Name!;
+
+            foreach (var author in authors)
+            {
```
Comment (katyalys) [inline MusicService/MusicService.Application/Services/ReleaseService.cs:222]: i guess you can use Any() instead of loop

**Review by katyalys — CHANGES_REQUESTED** [review]

File: MusicService/MusicService.API/Controllers/GenreController.cs (hunk 1 above)
Comment (AntonZhidovich) [reply MusicService/MusicService.API/Controllers/GenreController.cs:23]: removed

**Review by katyalys — APPROVED** [review]

**Pull request merged**